### PR TITLE
Refactoring

### LINKED ISSUE #3: Ability to pass custom line length
Instead of the -l option extending the line length to 99 make it possible to give it a custom line length.
implementation:
```
$ len8 -l=130 my_project
$ len8 -l 130 my_project
```

### LINKED ISSUE #14: Replace `argparse` with `click`
**Is your feature request related to a problem? Please describe.**
Switch to using `click` instead of `argparse` for the command parsing. This would allow for better command parsing as `click` is both more advanced and easier to use than `argparse`.

This would also rid len8 of it's "no dependencies" status, however as it is a development tool, this doesn't really matter.

### PR DESCRIPTION
**Pull request by parafoxia**
### Summary
This PR:

- Reimplements the CLI to use `click` rather than `argparse` (#14)
- Adds options for custom line lengths (#3)

### Tasks
- [x] I have searched for similar/duplicate PR's, and there are none.
- [x] I have made tests to validate any changes to the code base.
- [x] I have run `nox` and all sessions pass.

### Related issues
Closes #3.
Closes #14.

### CODE REVIEW

**Comment by parafoxia** [comment]
Don't worry about all the force pushes 😅

File: README.md [hunk 1]
```diff
@@ -61,9 +61,15 @@ len8 my_package tests
...
-len8 -l /home/project
+# Check 'project' dir and increase maximum allowed line lengths
+# Note that line lengths for comments and docs stay at 72
+len8 -l project         # Increase to 88 (black's default)
+len8 -ll /home/project  # Increate to 99 (max allowed by PEP 8)
+
+# Check using custom line lengths
+len8 -c 150 .
+len8 -d 100 .
+len8 -ll -d 99 .
```
Comment (Jonxslays) [inline README.md:72]:
```suggestion
len8 -ll /home/project  # Increase to 99 (max allowed by PEP 8)

# Check using custom line lengths
len8 -c 150 . # Increase code to 150
len8 -d 100 . # Increase docs to 100
len8 -ll -d 99 . # Increase code to 99, and docs to 99
```

File: README.md [hunk 2]
```diff
@@ -78,15 +84,17 @@ len8 -lx ignoreme.py ./project_dir
...
 checker = Checker(strict=True)
 
 # Set attributes after instantiation
-checker.extend = True
+checker.extend = 2
 checker.exclude = ["excluded_dir"]
 checker.strict = False
 
+# Set line lengths after instantiation
+checker.set_line_lengths(code=100, docs=80)
```
Comment (Jonxslays) [inline README.md:96]: In the code this method is called `set_lengths`

File: len8/checker.py [hunk 3]
```diff
@@ -36,13 +36,20 @@ class Checker:
...
-        exclude: list[pathlib.Path | str]
+        exclude: ``list[pathlib.Path | str]``
             A list of paths on top of the defaults (.nox, .venv, and
             venv) to exclude from checking. Defaults to an empty list.
-        extend: bool
-            Whether or not to increase acceptable line length to 99.
-            Defaults to ``False``.
-        strict: bool
+        extend: ``int``
+            Increase the code limit limit to set figures (pass ``1``
```
Comment (Jonxslays) [inline len8/checker.py:43]:
```suggestion
            Increase the line length limit to set figures (pass ``1``
```

File: len8/checker.py [hunk 4]
```diff
@@ -97,18 +115,48 @@ def exclude(self, excludes: t.List[Path]) -> None:
...
+        if not 0 <= extend <= 2:
+            raise ValueError("'extend' should be between 0 and 2 inclusive")
         self._extend = extend
 
+    @property
+    def code_length(self) -> int:
+        """The value to use as the maximum line length for code. This
+        will return:
+
+        - The custom length, if :obj:`_code_length` is not ``None``.
```
Comment (Jonxslays) [inline len8/checker.py:137]: I dont think predicating what this returns on a private attribute is a good idea. Maybe something like "if a custom length was set" or something.

File: len8/checker.py [hunk 5]
```diff
@@ -97,18 +115,48 @@ def exclude(self, excludes: t.List[Path]) -> None:
...
+            return self._code_length
+
+        return (79, 88, 99)[self._extend]
+
+    @property
+    def docs_length(self) -> int:
+        """The value to use as the maximum line length for comments and
+        documentation. This will return:
+
+        - The custom length, if :obj:`_docs_length` is not ``None``.
```
Comment (Jonxslays) [inline len8/checker.py:152]: Same as above in code_length.

File: len8/cli.py [hunk 6]
```diff
@@ -26,42 +26,74 @@
...
+        "Increase line length for code. Option is additive, and can be used "
+        "up to 2 times (corresponding to 88 and 99; omit for 79)"
+    ),
+)
+@click.option(
+    "-c",
+    "--code-length",
+    type=int,
+    metavar="CHARS",
+    help="Custom line length for code. Overrides --length if set.",
```
Comment (Jonxslays) [inline len8/cli.py:71]: `--length` is not a valid option, either change it above to `-l`, `--length`, or change it here to `--extend-length`

**Review by Jonxslays — CHANGES_REQUESTED** [review]
Couple of things but i love how you simplified `__main__`. :D

**Review by Jonxslays — APPROVED** [review]
nice

**Pull request merged**